File: nodeDepths.py
def nodeDepths(root,depth=0):
    if root is None:
        return 0
    return depth +nodeDepths(root.Left,depth+1) + nodeDepths(root.Right,depth=+1)


class BinaryTree:
    def __init__(self, value):
        self.value = value
        self.Left = None
        self.Right = None


root =BinaryTree(2)
root.Left= BinaryTree(3)
root.Right= BinaryTree(14)


print(nodeDepths(root))

File: NodeDepths.go
package main

import "fmt"

type Binary struct {
	Value       int
	Left, Right *Binary
}

func NodeDepths(root *Binary) int {
	return nodeDepthsHelper(root, 0)
}

func nodeDepthsHelper(root *Binary, depth int) int {
	if root == nil {
		return 0
	}
	return depth + nodeDepthsHelper(root.Left, depth+1) + nodeDepthsHelper(root.Right, depth+1)
}

func main() {

	newTree := Binary{1, nil, nil}
	newTree.Left = &Binary{2, nil, nil}
	newTree.Right = &Binary{3, nil, nil}
	newTree.Left.Left = &Binary{4, nil, nil}
	newTree.Left.Right = &Binary{5, nil, nil}
	newTree.Right.Left = &Binary{6, nil, nil}
	newTree.Right.Right = &Binary{7, nil, nil}
	fmt.Println(NodeDepths(&newTree))
}
